modules/http/http.c

```diff
@@ -261,6 +263,66 @@ http_dd_set_ssl_version(LogDriver *d, const gchar *value)
   return TRUE;
 }
 
+static inline gboolean
+_http_dd_curl_compression_string_match(const gchar *string, gint curl_compression_index)
+{
+  return (strcmp(string, curl_compression_types[curl_compression_index]) == 0);
+}
+
+gboolean
+_http_dd_check_curl_compression(const gchar *type)
+{
+  if(_http_dd_curl_compression_string_match(type, CURL_COMPRESSION_UNCOMPRESSED)) return TRUE;
+#if defined(CURL_COMPRESSION_IMPLEMENTED_GZIP) && defined(ZLIB_AVAILABLE)
```

Comment:
[optional]

I am reviewing this PR, and going commit-by-commit.

The `..._AVAILABLE` flags were used here prematurely, they will be introduced in a later commit, then in an even later commit another alternative gets introduced and gets used in this function.

A more comprehensible way of building up the git history would be to write the logic of setting the correct flag first (`SYSLOG_NG_HAVE_ZLIB`), then use it in the necessary places.

In the end you have arrived to a correct code, but the way of getting there could be improved in the git history.